tests/test_git_poller.py

```diff
@@ -1,10 +1,106 @@
 # Copyright (C) 2017 Linaro Limited
 # Author: Andy Doan <andy.doan@linaro.org>
+import random
+from typing import List
+from unittest import TestCase, mock
 
 from jobserv import git_poller
 
-from unittest import TestCase, mock
+def _fake_change_params(head_sha: str='fake', url: str='https://fake.com'):
+    return {
+        'GIT_OLD_SHA': f'old{head_sha}',
+        'GIT_SHA': head_sha,
+        'GIT_URL': url,
+    }
+
+def _fake_entry() -> git_poller.PollerEntry:
+    return git_poller.PollerEntry(
+        trigger=_fake_trigger(),
+        definition=git_poller.ProjectDefinition(
+            {"triggers": [], }))
+
+def _fake_trigger() -> git_poller.ProjectTrigger:
+    return git_poller.ProjectTrigger(
+        12, 't', 'proj', 'user', 1,)
 
+def _fake_github_content(
+    skip_head_content: bool=False, skip_other_content: bool=False,
+    head_sha: str="fakeheadsha") -> List[dict]:
+    msg = "fake %s fake\nmore fake"
+    choices = ('[skip ci]', '[ci skip]')
+    head_content = msg % (random.choice(choices) if skip_head_content else '')
+    other_content = msg % (random.choice(choices) if skip_other_content else '')
+    # truncated output below
+    return [
+      {
+        "sha": head_sha,
+        "commit": {
+          "message": head_content,
+        },
+      },
+      {
+        "sha": f"not{head_sha}",
+        "commit": {
+          "message": other_content,
+        },
+      }
+    ]
+
+def _fake_gitlab_content(
+        skip_head_title: bool=False, skip_head_content: bool=False,
+        skip_other_title: bool=False, skip_other_content: bool=False,
+        head_sha: str="fakeheadsha") -> List[dict]:
```

Comment:
since you are introducing strong typing you might as well make this `List[Dict[str, str]]` to be even more specific

Reply:
lol - force of habit.  I've updated it.  I would actually appreciate even stronger typing, but these are tests so I'm ok 😜